test/unit/switch.spec.ts

```diff
@@ -359,3 +365,61 @@ test("switch fallthrough stops after default", () => {
         return out;
     `.expectToMatchJsResult();
 });
+
+test("switch does not pollute parent scope", () => {
+    util.testFunction`
+        let x: number = 0;
+        let y = 1;
+        switch (x) {
+            case 0:
+                let y = 2;
+        }
+        return y;
+    `.expectToMatchJsResult();
+});
+
+test("switch collapses empty case and minimizes conditions", () => {
+    util.testFunction`
+        const out = [];
+        switch (5 as number) {
+            case 0:
+            case 1:
+            case 2:
+                out.push("0,1,2");
+            case 3:
+                out.push("3");
+                break;
+            default:
+                out.push("default");
+            case 4:
+                out.push("4");
+        }
+        return out;
+    `
+        .expectLuaToMatchSnapshot()
+        .expectToMatchJsResult();
+});
+
+test("switch handles side-effects", () => {
+    util.testFunction`
+        const out = [];
+        
+        let y = 0;
```

Comment:
It would be good to have variants of this test with different starting y values.